chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -207,6 +207,15 @@ void MdTextButton::SetLoading(bool loading) {
   UpdateColors();
 }
 
+void MdTextButton::UpdateBackgroundColor() {
+  // Handled via |UpdateColorsForBrave|.
+  if (kind_ != kOld) {
```

Comment:
How about splitting `UpdateColorsForBrave()` into two parts? Background part and other parts.
and moving Background parts into here to sync with upstream structure.
If upsteam changes to call `UpdateBackgroundColors()` from another places in the future, it could make regression?

Reply:
This is on my list, but it's a bit of a bigger change and I'd like to keep this as minimal as possible so we can get it uplifted. I don't think, upstream to calling `UpdateBackgroundColors` will break anything because it will use our colors (either the existing ones if `kind_ == kOld` or the new ones, otherwise).

Reply:
Start of a follow up PR is here: https://github.com/brave/brave-core/pull/16124